compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -1906,6 +1921,38 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
           newBody)(methodDef.optimizerHints, None)(methodDef.pos)
     }
 
+    private def patchTypesOfLocals(methodDef: js.MethodDef,
+        patches: mutable.Map[LocalName, ir.Types.Type]) = {
```

Comment:
This should be an immutable `Map`. It's never mutated.

Reply:
Also, it's usually better to explicitly specify the result type of a function that is a bit long, even when it is private. It helps the reader.